static/js/main.js

```diff
@@ -260,14 +268,31 @@ function flipImage(screenshot_id) {
     canvas.height = img.naturalHeight;
 
     let context = canvas.getContext('2d');
-    context.scale(-1, 1);
-    context.drawImage(img, -img.naturalWidth, 0);
 
-    // Draw the canvas to the image area.
-    img.src = canvas.toDataURL('image/jpeg');
+    // Horizontal flip.
+    if (axis === 'y') {
```

Comment:
I see a lot of duplicate code here.
You should be able to clean it up so that you just tweak some values for each case and use most of the same code.